pyvista/core/dataset.py

```diff
@@ -170,6 +170,7 @@ def __eq__(self: ActiveArrayInfo, other: object) -> bool:
         return False
 
 
+@promote_type(_vtk.vtkDataSet)
 @abstract_class
 class DataSet(DataSetFilters, DataObject):
```

Comment:
Highlighting this to demonstrate how the plugin works. Use the decorator to promote `DataSet` as a `vtkDataSet`
This is similarly done for `DataObject` and `vtkDataObject`